pkg/gadget-service/instance-manager/file-store/filestore.go

```diff
@@ -0,0 +1,185 @@
+// Copyright 2023-2024 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package filestore
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+
+	log "github.com/sirupsen/logrus"
+	"google.golang.org/protobuf/encoding/protojson"
+
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/gadget-service/api"
+	instancemanager "github.com/inspektor-gadget/inspektor-gadget/pkg/gadget-service/instance-manager"
+)
+
+var GadgetFilePath = "/var/lib/ig/gadgets"
+
+type FileStore struct {
+	api.GadgetInstanceManagerServer
+	instanceMgr *instancemanager.Manager
```

Comment:
The manager is basically doing the runtime things (can spawn gadgets and attach to them), while the store is what's reading and writing configurations. If applying a new configuration, the store is what triggers the manager to actually run a gadget.

So the store will control the manager, the manager doesn't however need to control the store. Since the store is part of the manager right now, I can see where the confusion is coming from.

Also, since a recent update, I made the manager to a proxy for the store calls in order to apply some general sanitizing independent of the store implementation.

I will split it more, so that the manager doesn't need a store but the store will have a reference to the manager.